python/TestHarness/schedulers/QueueManager.py

```diff
@@ -504,63 +303,254 @@ def augmentQueueParamsBase(self, job_container):
-        template = self.augmentQueueParams(job_container, template)
+        template = self.augmentQueueParams(job, template)
 
         return template
 
-    def waitFinish(self):
+    def reserveSlots(self, job):
         """
-        Inherited method called once all jobs have been sent to queueJobs.
-        QueueManager uses this method as a means to trigger other post-operation
-        methods after _all_ jobs have been launched. Such as calling the
-        queueJobs method a second time to perform status checks. As well as
-        writing the finalized json session file.
+        Inherited method which controls when jobs are allowed to execute,
+        depending on available resources.
+
+        QueueManager only executes third party queueing commands. So
+        modify every job to only require 1 process.
         """
+        job.setProcessors(1)
+        return Scheduler.reserveSlots(self, job)
 
-        # Set the status_check switch
-        if self.__status_check == False:
-            self.__status_check = True
+    def testOutput(self, job):
+        """
+        Adjust the Tester for a proper working directory (QueueManager creates sub-directores),
+        and allow derived Tester to perform processResults.
 
-            # TestHarness is finished sending us jobs, do postLaunch stuff (for PBS this
-            # means releasing the jobs using qrls)
-            self.postLaunch(self.__jobs)
+        Return resulting output.
+        """
+        tester = job.getTester()
+        json_job = self.getData(job.getUniqueIdentifier(), std_out=True, working_dir=True)
 
-            # Inform the user of our status change
-            print('\nJobs launched. Begin checking status...')
+        if json_job['std_out']:
+            stdout_file = os.path.join(json_job['working_dir'], json_job['std_out'])
+        else:
+            stdout_file=''
 
-            # Call queueJobs again using the populated set of jobs as our list (clever!
-            # no longer do we need to nerf the TestHarness's os.walk routine...)
-            self.queueJobs(status_jobs=self.__jobs)
+        if os.path.exists(stdout_file):
+            file = open(stdout_file, 'r+')
+            outfile = file.read()
+        else:
+            self.adjustJobStatus(job, tester.bucket_fail, 'NO STDOUT FILE')
+            return ''
 
-        # Write session data to file
-        self.writeSessionFile()
+        # Alter test_dir to reflect working_dir creation
+        tester.specs['test_dir'] = json_job['working_dir']
+
+        if tester.hasRedirectedOutput(self.options):
+            outfile += util.getOutputFromFiles(tester, self.options)
+
+        # Allow the tester to verify its own output.
+        output = tester.processResults(tester.specs['moose_dir'], self.options, outfile)
+
+        # Set the testers output with modifications made above
+        job.setOutput(output)
+
+        # Append processResults to our stdout file
```

Comment:
Either the comment, or the logic is wrong here. You say "append" but then you seek to the beginning of the fie clobbering whatever is already there.

Reply:
Well, I can probably word it better...
What we are trying to do here, is append the processResults output, to the file created by qsub (test_name.o12345)

```python
output = file.read(previous stdout generated by qsub)
output = tester.processResults(output)
file.seek(0)
file.write(output)
file.truncate()
file.close()
```
Keep in mind we only allow this to happen once. Now that the file is whole, with processResults output in there, we can do things like --verbose, --timing, etc...

Reply:
I'd just remove append then. I'd suggest saying that you are replacing the existing content with the output + status.